adam_core/coordinates/transform.py

```diff
@@ -1213,23 +1214,28 @@ def cartesian_to_origin(
     ----------
     coords : `~adam_core.coordinates.cartesian.CartesianCoordinates`
         Cartesian coordinates and optionally their covariances.
-    origin : {'SUN', 'SOLAR_SYSTEM_BARYCENTER'}
+    origin : `~adam_core.coordinates.origin.OriginCodes`
         Name of the desired origin.
 
     Returns
     -------
     CartesianCoordinates : `~adam_core.coordinates.cartesian.CartesianCoordinates`
         Translated Cartesian coordinates and their covariances.
     """
-    unique_origins = np.unique(coords.origin)
-    vectors = np.zeros(coords.values.shape, dtype=np.float64)
+    origins = coords.origin.code.to_numpy(zero_copy_only=False)
+    unique_origins = np.unique(origins)
```

Comment:
Since `zero_copy_only=False`, it's a good idea to do as much computation as possible in arrow-space:

```suggestion
    unique_origins = coords.origin.code.unique().to_numpy(zero_copy_only=False)
```

Reply:
The rewrite is a little more complicated since we use the original non-deduped origin array to create a mask: ```mask = np.where(origins == origin_in)[0]```, but I'm inspired. Let me try this fully with pyarrow.

Reply:
This https://github.com/B612-Asteroid-Institute/adam_core/pull/28/commits/33b886d793f47fd198f35d63e2933384073e3ac5 implementation still is not zero copy but its copying an array of booleans and not strings so hopefully that is better.